### ISSUE TITLE
support TextDiffuser pipelines natively in Diffusers

### ISSUE THREAD

**Issue by bghira** [huggingface/diffusers#8410]
### Model/Pipeline/Scheduler description

![image](https://github.com/microsoft/unilm/raw/master/textdiffuser/assets/readme_images/introduction.jpg)

TextDiffuser is an open-source framework and pretrained model series for generating unique depth maps that can be used to guide the model for typography.

It not only has open code and weights, but the dataset is also available, containing ~9.7 million typography samples 🤯 

### Open source status

- [X] The model implementation is available.
- [X] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

Original repository: https://github.com/microsoft/unilm/tree/master/textdiffuser
Follow-up repository: https://github.com/microsoft/unilm/tree/master/textdiffuser-2
Demo video (not mine): https://youtu.be/ApcJ1UyLQB8?t=62 (timestamp)

**Comment by bghira** [comment]
additionally they have a [training script](https://github.com/microsoft/unilm/blob/master/textdiffuser/train.py) based on the Diffusers example t2i script

**Comment by sayakpaul** [comment]
If they have built it on top of `diffusers`, then it's already a good sign. I think it might be a good idea for users to use their work instead. But will see and decide based on community interest. 

@yiyixuxu FYI.

**Comment by MnCSSJ4x** [comment]
From my understanding, we need to create a huggingface model/ a diffuser for this given model, right? If that's the case I would like to try this.
